### DISCUSSION TITLE
Breaking changes to be included in Mesa 2.0

### DISCUSSION THREAD

**Discussion by rht** [mesa/mesa#1511] [General]
I'm compiling a list of breaking changes we should do for Mesa 2.0.

- [ ] `SingleGrid.position_agent(agent, x, y)` should be deprecated in favor of `place_agent` and `move_to_empty`. `position_agent` is error-prone. E.g. A bug has existed in the Schelling example for a very long time, where the number of agents in the scheduler does not match the number of agents in the grid. This is only discovered recently and fixed in #1509.
- [ ] Remove all deprecated features and their warnings
  * [ ] Remove neighbor_iter #1687
  * [ ] Remove find_empty, num_agents in move_to_empty, position_agent #1688
- [x] `Grid.get_neighborhood` must raise an error when position is out of bound, where currently it silently returns an empty list https://github.com/projectmesa/mesa/pull/1476#issuecomment-1301238002. PR: #1524
- [ ] The current `get_neighbors` in `NetworkGrid` is more equivalent to `get_neighborhood` in the other grid classes. Implement `get_neighbors` and `get_neighborhood` that are consistent with other grid classes, as stated in #1542
- [ ] #1566
- [ ] Remove BatchRunner class

Debatable:
- Rename `model.grid` to `model.space` #1344

**Comment by ameligrana** [comment]
We should start to deprecate `empties` and in 2.0 create a `track_empties` variable defaulting to false but if enabled makes it possible to track the `empties` set as before.

Edit: not necessary anymore after #1546

**Comment by EwoutH** [comment]
Since those changes are already in the main branch, would that mean the next release will be major one (2.0) instead of a minor one (1.2)?

**Reply by rht** [reply]
The PRs linked didn't do any breaking changes. Next release is likely 1.2.

**Reply by EwoutH** [reply]
Ah right, they are only deprecations.

**Comment by jackiekazil** [comment]
I believe this is one, yes? https://github.com/projectmesa/mesa/pull/1524

**Reply by rht** [reply]
Yes, I updated the list.
